rootfs/api/models/config.py

```diff
@@ -137,13 +111,12 @@ def save(self, **kwargs):
                         # error if unsetting non-existing key
                         if key not in data:
                             raise UnprocessableEntity('{} does not exist under {}'.format(key, attr))  # noqa
-
                         data.pop(key)
                     else:
                         data[key] = value
                 setattr(self, attr, data)
 
-            self.set_healthchecks()
+            self._migrate_legacy_healthcheck()
```

Comment:
Nice! We just need to aggressively ensure that people don't have legacy health check and then try to set new checks. On controller restart it would always overwrite. Maybe add in a "if livenessProbe in self.healthcheck" is true then skip setting legacy